### PR TITLE
Added FCFS Scheduling

### LINKED ISSUE #1046: Adding FCFS (First Come First Served) Scheduling Algorithm in C++, Python, Java, Golang
Please assign it to me.

### PR DESCRIPTION
**Pull request by MastersAbh**
Fixes #1046  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added FCFS Scheduling Algorithm in C++, Java, Python & Golang

#### Languages Used:
- C++
- Java
- Python
- Golang

#### Files Added:
- FCFS_Scheduling.cpp
- FCFS_Scheduling.java
- FCFS_Scheduling.go
- FCFS_Scheduling.py

### CODE REVIEW

File: FCFS_Scheduling/FCFS_Scheduling.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    */
+    for(lint i = 1; i < no_of_process; i++){
+        wait_time[i] = wait_time[i-1] + burst_time[i-1];
+    }
+    for(lint i = 0; i < no_of_process; i++){
+        turnaround_time[i] = burst_time[i] + wait_time[i];
+    }
+    for (int i = 0; i < no_of_process ; i++) {
+      cout << "Process " << i + 1 << " -->  (B.T) : " << burst_time[i] << " (W.T.) : " << wait_time[i] << " (T.A.T) : " << turnaround_time[i] << endl;
```
Comment (singh-shreya6) [inline FCFS_Scheduling/FCFS_Scheduling.cpp]: Please split this statement across multiple lines

File: FCFS_Scheduling/FCFS_Scheduling.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+}
+
+
+int main(){
+    lint no_of_process = 3;
+    lint burst_time[] = {5, 7, 3};
+    solve(no_of_process, burst_time);
+}
+/*
```
Comment (singh-shreya6) [inline FCFS_Scheduling/FCFS_Scheduling.cpp]: Please leave a blank line here.

File: FCFS_Scheduling/FCFS_Scheduling.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    for i in range(0, no_of_process) :
+        turnaround_time.append(burst_time[i] + wait_time[i])
+    for i in range(0, no_of_process) :
+        print("Process " , i + 1 , " -->  (B.T) : " , burst_time[i] , " (W.T.) : " , wait_time[i] , " (T.A.T) : " , turnaround_time[i])
+
+
+no_of_process = 3
+burst_time = [5, 7, 3]
+solve(no_of_process, burst_time)
+
```
Comment (singh-shreya6) [inline FCFS_Scheduling/FCFS_Scheduling.py]: One line gap is sufficient.

File: FCFS_Scheduling/FCFS_Scheduling.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+void solve(lint no_of_process, lint burst_time[]){
+    lint wait_time[no_of_process], turnaround_time[no_of_process];
+    wait_time[0] = 0; // wait time of 1st process is 0.
+    /*
+        We use the following equations for computing the result:
+
+        Wait time of process i = wait time of process i-1 + burst time of process i-1.
+        Turn Arond time of process i = burst time of process i + wait time of process i.
+
```
Comment (singh-shreya6) [inline FCFS_Scheduling/FCFS_Scheduling.cpp]: Blank line here is not necessary

File: FCFS_Scheduling/FCFS_Scheduling.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        for(int i = 0; i < no_of_process; i++){
+            turnaround_time[i] = burst_time[i] + wait_time[i];
+        }
+        for (int i = 0; i < no_of_process ; i++) {
+        System.out.println("Process " + (i + 1) + " -->  (B.T) : " + burst_time[i] + " (W.T.) : " + wait_time[i] + " (T.A.T) : " + turnaround_time[i]);
+        }
+    }
+
+
```
Comment (singh-shreya6) [inline FCFS_Scheduling/FCFS_Scheduling.java]: One line gap between different functions is sufficient.

File: FCFS_Scheduling/FCFS_Scheduling.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    wait_time = append(wait_time, 0) // wait time of 1st process is 0.
+    /*
+        We use the following equations for computing the result:
+
+        Wait time of process i = wait time of process i-1 + burst time of process i-1.
+        Turn Arond time of process i = burst time of process i + wait time of process i.
+
+    */
+    for i := 1; i <= no_of_process; i++ {
+        wait_time = append(wait_time, wait_time[i-1] + burst_time[i-1])
```
Comment (singh-shreya6) [inline FCFS_Scheduling/FCFS_Scheduling.go]: Please add space around operators i.e "-"

**Comment by MastersAbh** [comment]
Made changes, please review @singh-shreya6 .

File: FCFS_Scheduling/FCFS_Scheduling.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        Turn Arond time of process i = burst time of process i + wait time of process i.
+    */
+    for i := 1; i <= no_of_process; i++ {
+        wait_time = append(wait_time, wait_time[i - 1] + burst_time[i - 1])
+    }
+    for i := 0; i < no_of_process; i++ {
+        turnaround_time = append(turnaround_time, burst_time[i] + wait_time[i])
+    }
+    for i := 0; i < no_of_process ; i++ {
+		fmt.Println("Process " + strconv.Itoa(i + 1) + " -->  (B.T) : " + strconv.Itoa(burst_time[i]) +
```
Comment (singh-shreya6) [inline FCFS_Scheduling/FCFS_Scheduling.go]: Fix the indentations at print statements and your code would be ready to merge.

**Comment by MastersAbh** [comment]
Done. @singh-shreya6

**Comment by somya-kapoor** [comment]
@MastersAbh FCFS is already implemented in C++ in the repository.(https://github.com/jainaman224/Algo_Ds_Notes/blob/master/Scheduling_Algorithm/C%2B%2B/Non_Preemptive/First_Come_First_Served.cpp)
Also, change the file name accordingly.

**Comment by MastersAbh** [comment]
@somya-kapoor , I guess it is better to have a new folder for this PR, because the last one seems too similar with codes available at GFG, while this is a fresh code. @jainaman224

**Pull request merged**